lib/px/utils/config_loader.lua

```diff
@@ -0,0 +1,94 @@
+    local px_logger = require("px.utils.pxlogger").load(config_file)
+    px_logger.debug("Fetching configuration")
+    local cjson = require "cjson"
+    local px_server = config.configuration_server
+    local px_port = config.configuration_server_port
+    local path = px_constants.REMOTE_CONFIGURATIONS_PATH
+    local checksum = config.checksum
+    local query
+    if checksum ~= nil then
+        query = '?checksum=' .. checksum
+    else
+        query = ''
+    end
+
+    local httpc = http.new()
+    local ok, err = httpc:connect(px_server, px_port)
+    if not ok then
+        px_logger.error("HTTPC connection error: " .. err)
+    end
+    if config.ssl_enabled == true then
+        local session, err = httpc:ssl_handshake()
+        if not session then
+            px_logger.error("HTTPC SSL handshare error: " .. err)
+        end
+    end
+    local res, err = httpc:request({
+        path = path,
+        method = "GET",
+        headers = {
+            ["Content-Type"] = "application/json",
+            ["Authorization"] = "Bearer " .. config.auth_token
+        },
+        query = query
+    })
+
+    if err ~= nil or res == nil or res.status > 204 then
+        px_logger.error("Failed to get configurations: " .. (err ~= nil and err or ''))
+        if (checksum == nil) then --no configs yet and can't get configs - disable module
+            px_logger.error("Disabling PX module since no configuration is available")
+            config.px_enabled = false
+        end
+        return
+    end
+    if res.status == 204 then
+        px_logger.debug("Configuration was not changed")
+        return
+    end
+    -- new configurations available
+    if res.status == 200 then
+        local body = res:read_body()
+        px_logger.debug("Applying new configuration: " .. body)
+        body = cjson.decode(body)
+        config.checksum = body.checksum
+        config.px_enabled = body.moduleEnabled
+        config.cookie_secret = body.cookieKey
+        config.px_appId = body.appId
+        config.blocking_score = body.blockingScore
+        config.sensitive_headers = body.sensitiveHeaders
+        config.ip_headers = body.ipHeaders
+        config.px_debug = body.debugMode
+        config.block_enabled = body.moduleMode ~= "monitoring"
+        config.client_timeout = body.connectTimeout
+        config.s2s_timeout = body.riskTimeout
+        config.report_active_config = true
+   end
+end
+
+function _M.load(config_file)
+    local config = require(config_file)
+    local ngx_timer_at = ngx.timer.at
+    local px_logger = require("px.utils.pxlogger").load(config_file)
+    -- set interval
+    local function load_on_timer()
+        local ok, err = ngx_timer_at(config.load_interval, load_on_timer)
+        if not ok then
+            px_logger.error("Failed to schedule submit timer: " .. err)
+            px_logger.error("Disabling PX module since timer failed")
+            config.px_enabled = false
```

Comment:
that looks like a mistake. I don't think we need to disable the enforcer on timer error. only if it's the first time and we don't have any configurations yet